src/Tools/Common/Commands/ProcessStatus.cs

```diff
@@ -80,5 +180,351 @@ private static Process GetProcessById(int processId)
                 return null;
             }
         }
+
+        private static string GetArgs(Process process)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                try
+                {
+                    string commandLine = WindowsProcessExtension.GetCommandLine(process);
+                    if (!String.IsNullOrWhiteSpace(commandLine))
+                    {
+                        string[] commandLineSplit = commandLine.Split(' ');
+                        if (commandLineSplit.FirstOrDefault() == process.ProcessName)
+                        {
+                            return String.Join(" ", commandLineSplit.Skip(1));
+                        }
+                        return commandLine;
+                    }
+                }
+                catch (Exception ex) when (ex is Win32Exception || ex is InvalidOperationException)
+                {
+                    return "[Elevated process - cannot determine command line arguments]";
+                }
+
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                try
+                {
+                    string commandLine = File.ReadAllText($"/proc/{process.Id}/cmdline");
+                    if(!String.IsNullOrWhiteSpace(commandLine))
+                    {
+                        //The command line may be modified and the first part of the command line may not be /path/to/exe. If that is the case, return the command line as is.Else remove the path to module as we are already displaying that.
+                        string[] commandLineSplit = commandLine.Split('\0');
+                        if (commandLineSplit.FirstOrDefault() == process.MainModule?.FileName)
+                        {
+                            return String.Join(" ", commandLineSplit.Skip(1));
+                        }
+                        return commandLine.Replace("\0", " ");
+                    }
+                    return "";
+                }
+                catch (IOException)
+                {
+                    return "[cannot determine command line arguments]";
+                }
+            }
+            return "";
+        }
+    }
+
+    internal static class WindowsProcessExtension
+    {
+
+        //https://github.com/projectkudu/kudu/blob/787c893a9336beb498252bb2f90a06a95763f9e9/Kudu.Core/Infrastructure/ProcessExtensions.cs#L562-L616
```

Comment:
Duplicating code from other OSS projects requires being careful of license requirements out of courtesy to the original authors and to honor our legal obligations. Many licenses, such as the one used by the Kudu project, require us to preserve Kudu's original copyright and license notice. https://github.com/projectkudu/kudu/blob/787c893a9336beb498252bb2f90a06a95763f9e9/LICENSE.txt.

Check out https://github.com/dotnet/diagnostics/blob/main/THIRD-PARTY-NOTICES.TXT where we have been adding these notices.

(I'm not certain the attribution is actually required in this case, but its good habit and adding the attribution out of caution is probably less work than getting the exact answer)

Reply:
I created a pull request: https://github.com/dotnet/diagnostics/pull/2860